packages/transaction-controller/src/gas-flows/OptimismLayer1GasFeeFlow.ts

```diff
@@ -0,0 +1,84 @@
+import { Contract } from '@ethersproject/contracts';
+import { Web3Provider, type ExternalProvider } from '@ethersproject/providers';
+import { createModuleLogger, type Hex } from '@metamask/utils';
+
+import { CHAIN_IDS } from '../constants';
+import { projectLogger } from '../logger';
+import type {
+  Layer1GasFeeFlow,
+  Layer1GasFeeFlowRequest,
+  Layer1GasFeeFlowResponse,
+  TransactionMeta,
+} from '../types';
+import { buildUnserializedTransaction } from '../utils/layer1-gas-fee-flow';
+
+// This gas flow to be used for the following OP stack chains
+const OPTIMISM_STACK_CHAIN_IDS: Hex[] = [
+  CHAIN_IDS.OPTIMISM,
+  CHAIN_IDS.OPTIMISM_TESTNET,
+  CHAIN_IDS.BASE,
+  CHAIN_IDS.BASE_TESTNET,
+  CHAIN_IDS.OPBNB,
+  CHAIN_IDS.OPBNB_TESTNET,
+  CHAIN_IDS.ZORA,
+];
+
+const log = createModuleLogger(projectLogger, 'optimisim-layer1-gas-fee-flow');
+
+// Snippet of the ABI that we need
+// Should we need more of it at some point, the full ABI can be found here:
+// https://github.com/ethereum-optimism/optimism/blob/develop/gas-oracle/abis/OVM_GasPriceOracle.json
+const OPTIMISM_GAS_PRICE_ORACLE_ABI = [
+  {
+    inputs: [{ internalType: 'bytes', name: '_data', type: 'bytes' }],
+    name: 'getL1Fee',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+// BlockExplorer link: https://optimistic.etherscan.io/address/0x420000000000000000000000000000000000000f#code
+const OPTIMISM_GAS_PRICE_ORACLE_ADDRESS =
+  '0x420000000000000000000000000000000000000F';
+
+/**
+ * Optimism layer 1 gas fee flow that obtains gas fee estimate using Optimisim oracle contract.
+ */
+export class OptimismLayer1GasFeeFlow implements Layer1GasFeeFlow {
+  matchesTransaction(transactionMeta: TransactionMeta): boolean {
+    return OPTIMISM_STACK_CHAIN_IDS.includes(transactionMeta.chainId);
+  }
+
+  async #getOptimisimLayer1GasFee(
```

Comment:
Minor, but should we put the private methods at the bottom?

Reply:
Fixed